### DISCUSSION TITLE
Suricata custom address groups/modifying rules

### DISCUSSION THREAD

**Discussion by Hammer-Bear** [Security-Onion-Solutions/securityonion#5165] [Unsupported Versions] [answered]
Hey, so I come from a snort background so the suri stuff while I know it's not too different I'm having trouble making sense of it while in a SO environment.
So we have a monitor server that is generating false positives (and a lot of them) in snort I could add a variable to this rule to basically say if it matches this var ignore it i.e. `alert tcp !$VAR_NAME, any ->...` i know this is basically how address groups work but what I'm struggling to do is add it to the rules affected as it looks like so-rule modify replaces? and I don't want to replace it, just add that address group into the sources. If anyone can point me in the right direction would be greatly appreciated.

**Comment by gcsrellis** [comment]
You can edit the /opt/so/saltstack/local/pillar/global.sls on the manager node to specify suricata variables that will work on rules.  Then you can use the appropriate modify rules or thresholds to tune your alerts referencing the variable like you mentioned. 

suricata;
&nbsp;&nbsp;config;
&nbsp;&nbsp;&nbsp;&nbsp;vars:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;address-groups:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;HTTP_SERVERS: '[8.8.8.8]'
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;CUSTOM_NAME: '[8.8.8.8]'

**Reply by Hammer-Bear** [reply]
I understand that part, the editing of the actual rules as from what I can tell the so-rule modify replaces and I don’t want to replace I just want to add a variable to the source so it will still trigger but ignore any ip in that var group. so it looks like `alert tcp $HOME_NET, !$VAR any -> …`

**Reply by Mfisher925** [reply]
Did you successfully add a new variable?  Also were you able to modify the rule?  Or did you go with the threshold suppress method?

**Comment by gcsrellis** [comment, marked as answer]
You can use thresholding to suppress the alert from triggering from any IP in the variable like this:

thresholding:
&nbsp;&nbsp;sids:
&nbsp;&nbsp;&nbsp;&nbsp;2027201:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;- suppress:
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;gen_id: 1
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;track: by_dst
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ip: CUSTOM_VARIABLE

**Reply by Hammer-Bear** [reply]
Thanks, i understand it all now. I've gone through and made the suppressions but now when i run `sudo salt \* state.highstate` i get the following error on the nodes "`Rendering SLS 'global' failed. Please see master log for details.`" i would look at this "master log" but i dont know which log it is referring to. im assuming its some sort of formatting error but i can't see exactly where the formatting error is even if I copy from the templates.

**Reply by gcsrellis** [reply]
It is most likely a format error. Be mindful of the white spaces. 2 additional spaces on subsequent indented lines. There is a 64 character limit for the ip field that also applies to variables.  If you are using a variable maybe use a singular threshold with an ip to test your syntax first.  Also pasting the syntax into the global.sis sometimes messes with the formatting even though it may look correct. Try typing in the rule.

**Reply by Hammer-Bear** [reply]
yep was formatting, one space missing lol, thanks for all the assistance.
